File: app/src/main/java/linusfessler/alarmtiles/ComponentStateController.java
package linusfessler.alarmtiles;

import android.content.ComponentName;
import android.content.Context;
import android.content.pm.PackageManager;

public class ComponentStateController {

    private final Context context;
    private final PackageManager packageManager;

    public ComponentStateController(final Context context) {
        this.context = context.getApplicationContext();
        this.packageManager = context.getPackageManager();
    }

    public void enable(final ComponentName componentName) {
        packageManager.setComponentEnabledSetting(componentName,
                PackageManager.COMPONENT_ENABLED_STATE_ENABLED,
                PackageManager.DONT_KILL_APP);
    }

    public void disable(final ComponentName componentName) {
        packageManager.setComponentEnabledSetting(componentName,
                PackageManager.COMPONENT_ENABLED_STATE_DISABLED,
                PackageManager.DONT_KILL_APP);
    }

    public void enable(final Class cls) {
        final ComponentName componentName = new ComponentName(context, cls);
        enable(componentName);
    }

    public void disable(final Class cls) {
        final ComponentName componentName = new ComponentName(context, cls);
        disable(componentName);
    }

    public void enable(final String cls) {
        final ComponentName componentName = new ComponentName(context, cls);
        enable(componentName);
    }

    public void disable(final String cls) {
        final ComponentName componentName = new ComponentName(context, cls);
        disable(componentName);
    }
}

File: app/src/main/java/linusfessler/alarmtiles/ComponentStateController.kt
package linusfessler.alarmtiles

import android.app.Application
import android.content.ComponentName
import android.content.pm.PackageManager
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class ComponentStateController
@Inject
constructor(private val application: Application) {
    private val packageManager: PackageManager = application.packageManager

    fun enable(componentName: ComponentName) {
        packageManager.setComponentEnabledSetting(componentName,
                PackageManager.COMPONENT_ENABLED_STATE_ENABLED,
                PackageManager.DONT_KILL_APP)
    }

    fun disable(componentName: ComponentName) {
        packageManager.setComponentEnabledSetting(componentName,
                PackageManager.COMPONENT_ENABLED_STATE_DISABLED,
                PackageManager.DONT_KILL_APP)
    }

    fun enable(cls: Class<*>) {
        val componentName = ComponentName(application, cls)
        enable(componentName)
    }

    fun disable(cls: Class<*>) {
        val componentName = ComponentName(application, cls)
        disable(componentName)
    }

    fun enable(cls: String) {
        val componentName = ComponentName(application, cls)
        enable(componentName)
    }

    fun disable(cls: String) {
        val componentName = ComponentName(application, cls)
        disable(componentName)
    }
}